salt/states/file.py

```diff
@@ -677,6 +677,16 @@ def _delete_not_kept(nfn):
                 os.path.relpath(nfn, root), None, exclude_pat
             ):
                 return
+            # Before we can accurately assess the removal of a file, we must
+            # check for windows case sensitive files. If we originally meant
+            # to keep a file, but due to case sensitivity python would otherwise
+            # remove the file, check against the original list.
+            if win_keep:
+                for item in win_keep:
+                    if item.lower() != nfn.lower():
```

Comment:
```suggestion
                    if item.casefold() != nfn.casefold():
```

Reply:
Fyi, `casefold` is Unicode aware, ie, in some languages, I think Russian, some uppercased letter don't have the same meaning as it's lowercase.
And case fold handles those.

Reply:
Nice, never seen that function before ill use this + wayne's suggestion into one